src/toplevel.pl

```diff
@@ -96,6 +109,8 @@
     write('Print version information and exit'), nl,
     write('   -g, --goal GOAL        '),
     write('Run the query GOAL'), nl,
+    write('   -t GOAL                '),
+    write('Use GOAL as custom toplevel (arity 0 predicate)'), nl,
```

Comment:
It is a **goal**, the invoked predicate can have any arity.

For instance, if you only need an error code and not the type of error, you can use:

<pre>
$ scryer-prolog <b>-t 'halt(1)'</b> -g run,halt your_file.pl
</pre>

Does that work in your scripts? In that way, you do not have to implement a custom toplevel for this particular purpose.